### PR TITLE
RavenDB #6737 + RavenDB #6738

### PR DESCRIPTION
**Pull request by mobilunity-user**

### CODE REVIEW

File: test/session_tests/DocumentQueryTest.ts [hunk 1]
```diff
@@ -24,222 +28,181 @@ describe('Document query test', () => {
...
-      });
+  beforeEach(async () => {
+    store = DocumentStore.create(defaultUrl, defaultDatabase).initialize();
+    session = store.openSession();
+
+    const requestsExecutor: RequestsExecutor = store.getRequestsExecutor();
+    const productsTestingSort: ProductsTestingSort = new ProductsTestingSort(requestsExecutor);
+
+    await productsTestingSort.execute();
+    await session.store<Product>(session.create<Product>(new Product('products/101', 'test101', 2, 'a')));
```
Comment (ayende) [inline test/session_tests/DocumentQueryTest.ts:39]: `session.create<Product>` Why is this needed?

Comment (mobilunity-user) [reply test/session_tests/DocumentQueryTest.ts:39]: @ayende http://issues.hibernatingrhinos.com/issue/RavenDB-6745

Comment (gregolsky) [reply test/session_tests/DocumentQueryTest.ts:39]: @mobilunity-user I'm not sure I follow. We just wanted to have generic load and query methods. I can't recall anything about create(). Could you elaborate please?
